lib/rex/socket/range_walker.rb

```diff
@@ -540,8 +540,11 @@ def ipv6?
 class Host < Range
   attr_accessor :hostname
 
-  def initialize(address, hostname=nil, options=nil)
-    address = Rex::Socket.addr_atoi(address) if address.is_a? String
+  def initialize(address, hostname=nil, options={})
+    if address.is_a? String
+      options.merge!({ ipv6: address.include?(":") }) if options[:ipv6].nil?
```

Comment:
If this is intended to check if `address` is an IPv6 address, then we should be using the [`is_ipv6?`](https://github.com/rapid7/rex-socket/blob/e5a99ec35d13f64aefc0c76e849f77a39688ad55/lib/rex/socket.rb#L134) function instead.

Reply:
Copied the existing pattern - my guess is that its done this way for speed - much faster than a regex match. Happy to change it if you'd like.